include/tester/ssp/Softphone.hpp

```diff
@@ -1,15 +1,72 @@
 #pragma once
 
+#include<iostream>
+#include<string>
+#include<functional>
+#include<memory>
+
 #include"SSPAccount.hpp"
 #include"SSPCall.hpp"
+#include"SoftphoneArguments.hpp"
 
 class Softphone
 {
 public:
-    Softphone(/* args */);
-    ~Softphone();
+    Softphone(const SoftphoneArguments & args):
+        _account(args.id, args.domain, args.secret, std::bind(&Softphone::onIncomingCall, this, std::placeholders::_1))
+    {
+        _call = std::make_shared<SSPCall>(&_account, std::bind(&Softphone::onCallState, this,
+            std::placeholders::_1, std::placeholders::_2));
+        _account.apply();
+    }
 
-private:
-    std::vector<SSPAccount *> _accounts;
+    ~Softphone() = default;
+
+    void call(const std::string & uri)
+    {
+        _call->callTo(uri);
+    }
+
+    void onCallState(pj::CallInfo ci, const pj::OnCallStateParam &prm)
+    {
+        PJ_UNUSED_ARG(prm);
+        std::cout << "*** Call: " <<  ci.remoteUri << " [" << ci.stateText
+        << "]" << std::endl;
+        if (ci.state == PJSIP_INV_STATE_DISCONNECTED) {
+            /* Delete the call */
+            _call.reset();
+        }
+    }
 
-};
+    void onIncomingCall(const pj::OnIncomingCallParam &iprm)
+    {
+        std::shared_ptr<SSPCall> in_call = std::make_shared<SSPCall>(&_account, std::bind(&Softphone::onCallState,
```

Comment:
Great job! Now, since `std::make_shared` returns `std::shared_ptr` indefinitely, you can use the `auto` keyword instead of explicitly specifying the type.

Btw rename the variable to use camelCase lol

Reply:
done